atom/browser/ui/file_dialog_gtk.cc

```diff
@@ -169,14 +169,15 @@ void FileChooserDialog::OnFileDialogResponse(GtkWidget* widget, int response) {
 
   if (!save_callback_.is_null()) {
     if (response == GTK_RESPONSE_ACCEPT)
-      save_callback_.Run(true, GetFileName());
+      save_callback_.Run(true, GetFileName(), "");
     else
-      save_callback_.Run(false, base::FilePath());
+      save_callback_.Run(false, base::FilePath(), "");
   } else if (!open_callback_.is_null()) {
     if (response == GTK_RESPONSE_ACCEPT)
-      open_callback_.Run(true, GetFileNames());
+      open_callback_.Run(true, GetFileNames(), std::vector<std::string>());
     else
-      open_callback_.Run(false, std::vector<base::FilePath>());
+      open_callback_.Run(false, std::vector<base::FilePath>(),
+                         std::vector<std::string>());
```

Comment:
All these look awful.
Since that bookmark stuff is Mac-specific `OpenDialogCallback` and `SaveDialogCallback` should probably be different on different platforms.